File: Data Structres/Queue.cpp
#include<stdio.h>
#define MAX 10
int QUEUE[MAX],front=-1,rear=-1;
void inserq(int num);
int delq();
int showq();
int main()
{
	int item,choice;
	do
	{
		printf("\n\tQueue Using Array\n\n\t1.Insert\n\t2.Delete\n\t3.Display\n\t4.Exit\n\tEnter Ypur Choice");
		scanf("%d",&choice);
		switch(choice)
		{
			case 1:printf("\nEnter Data To Insrt:");
			scanf("%d",&item);
			insertq(item);
			break;
			case 2:printf("Deleted Data Is:%d",delq());
			break;
			case 3:showq()
			break;
			case 4:break;
			default:printf("Enter The Correct Choice:");
		}
			}while(choice!=4);
}
void insertq(int num)
{
if(rear=MAX-1)
{
	printf(Queue Overflow);
	return 0;
	}
	if(rear==-1)
	++front;
	queue[++rear]=num;
}
void delq()
{
	int item;
	if(front==-1)
	{
		printf("\nQueue Underflow");
		return 0;
	}
	item=queue[front];
	if(front==rear)
	front=rear-1;
	else
	++front;
	return item;
}
int showq()
{
	int i;
	if(front==-1)
	{
		printf("\nNo Element Found in Queue!");
		return 0;
	}
	printf("\nThe Queue Is:");
	for(i=front;i<=rear;i++)
	printf("%d",QUEUE[i]);
	return 0;
}

File: Data Structres/Queue.java
// Stack using array in Java
class Queue {
	int arr[];
	int front, rear;
	Queue(int size) {
		arr = new int[size];
		rear = -1;
		front = -1;
	}
void enqueue(int item) {
	if(front == arr.length -1)
		System.out.println("Stack overflow");
	else {
		if(front < 0){
			front = 0;
		}
		rear++;
		arr[rear] = item;
	}
}
	int dqueue() {
		int element;
		if(front < 0) {
			System.out.println("Queue is empty");
			return 0;
			}
		else
		{	
			element = arr[front];
			if(front >= rear)
			{
				rear = -1;
				front = -1;
			}
			front++;
			return element;
		}
}
}
class testQueue {
	public static void main(String arg[]) {
		Queue s1 = new Queue(6);
		Queue s2 = new Queue(8);
		s1.enqueue(10);
		s1.enqueue(15);
		s1.enqueue(20);

		System.out.println(s1.dqueue());
		System.out.println(s1.dqueue());
		System.out.println(s1.dqueue());
		System.out.println(s1.dqueue());

		for(int i = 0; i < 6; i++)
		{
			s2.enqueue(i);
			System.out.println(s2.dqueue());
		}
}
}
		